spec/operators/throttle-spec.ts

```diff
@@ -381,62 +381,6 @@ describe('throttle', () => {
     });
   });
 
-  it('should throttle by promise resolves', () => {
```

Comment:
After the changes, this test started failing. Reading the test, it should have *never* passed (as it's mixing a synchronous `TestScheduler` run with `Promise` resolution). Handling of promises (via converting `ObservableInput`s) is well-tested elsewhere, and we can just remove this test, and the test after it, as they are poorly written.

It's a bit more concerning to me that this test was passing before, and I'm unsure what made it finally behave correctly.

Reply:
@benlesh I had a quick look and this and for your peace of mind:

-   The test - as you mentioned - has an entirely synchronous source and test's expectation fails and throws.
-   The failures occurs in the call to the `complete` callback passed to `subscribe`, but the error bubbles all of the way back to the `throttle` operator's `onComplete` callback - which has a `try`/`catch` around it.
-   The `try`/`catch` around `onComplete` catches the error.
-   However, the `destination` is stopped, so the error is swallowed - well, the notification will be passed to the `onStoppedNotification` handler if there is one.

In the new implementation, the error is caught in the `SafeSubscriber` and it's passed to the `onUnhandledError`, instead.